File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = 'Yes'
    p1=0
    p2=0
    for i in goal :
        
        if p1 < len(cards1) and i in cards1[p1] :
            print(cards1[p1])
            p1+=1
            continue
            
        elif p2 < len(cards2) and i in cards2[p2] :
            print(cards2[p2])
            p2+=1
            continue
            
        else :
            answer = 'No'
            break
        
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        int point1 = 0;
        int point2 = 0;
        
        for (String goal_s : goal) {
            if(point1 < cards1.length && cards1[point1].equals(goal_s)){
                point1++;
            } else if(point2 < cards2.length && cards2[point2].equals(goal_s)){
                point2++;
            } else{
                answer = "No";
                break;
            }
        }
        
        return answer;
    }
}